Automattic/wp-calypso
# wpcom-block-editor: fix tracking events not being triggered

Our table for tracking `wpcom_pattern_inserted` Gutenberg pattern usage is showing 0 pattern insertions.

From from 24th May the event decreases and then plummets.

The issue affects the redux actions insertBlock and insertBlocks when inserting a block or a pattern, the action being called is the original rather than the rewritten one.

The API usage seems to be changed via [this](https://github.com/WordPress/gutenberg/pull/61472/files) with the unlock function. And it causes the previous way never works.

I got help to [revert](https://github.com/WordPress/gutenberg/pull/64063) the way it was working before. But it was closed since it promotes creating separate store subscriptions for private states. Furthermore I could easily happen again by using unlock in the future.

This PR avoids the spread of the actions and updates the condition for rewriting them to fix the issue.


Related to #
https://github.com/Automattic/wp-calypso/issues/92945
p1721705925726979-slack-C048CUFRGFQ
p1721791447737549-slack-CBTN58FTJ


## Proposed Changes

* Update how tracking actions are rewritten

## Why are these changes being made?


* We are loosing lots of tracking data since may

## Testing Instructions



* Cd to `/apps/wpcom-block-editor`
* Run `yarn dev --sync
* Open the editor 
* Setup the filter `t.gif` for the network requests 
* Verify if there is a tracking request when inserting a block
* Verify if there is a tracking request when inserting a pattern

## Pre-merge Checklist
